### ISSUE TITLE
Section links in markdown mainpage not working

### ISSUE THREAD

**Issue by bollenn** [doxygen/doxygen#8956]
**Describe the bug**

Linking to header labels in MD Mainpage in Latex format not working.

```md
## Header 1

## Header 2 {#label_header_2}

[Link to header 1](#header-1)

[Link to header 2](#label_header_2)
```
will be translated in Latex to:

```latex
\hypertarget{index_autotoc_md1}{}\doxysection{Header 1}\label{index_autotoc_md1}

\hypertarget{index_label_header_2}{}\doxysection{Header 2}\label{index_label_header_2}

\href{\#header-1}{\texttt{ Link to header 1}}

\href{\#label_header_2}{\texttt{ Link to header 2}} 
```

in HTML is becomes:

```html
<div class="textblock"><h1><a class="anchor" id="autotoc_md1"></a>
Header 1</h1>
<h1><a class="anchor" id="label_header_2"></a>
Header 2</h1>
<p><a href="#header-1">Link to header 1</a></p>
<p><a href="#label_header_2">Link to header 2</a> </p>
```

1. markdown headers do not get proper labels but labels as `<filename>_autotoc_md<index>` and as such it is hard or impossible for a user to link to it
2. It seems the Latex writer is prepending the label with `<filename>_` and this makes the link not work in the end. the HTML writer does not do this.

**Expected behavior**
1. markdown headers should get proper labels
2. Latex writer shall not edit labels

**Screenshots**
None

**To Reproduce**
See above

**Version**
latest master, 1.9.3 (8f7d18e59b8ede82077a9a14ca12bb9a0c565f51)

**Stack trace**
None

**Additional context**
None

**Comment by albert-github** [comment]
Please specify full version of "latest master" as this is a moving target (easiest `doxygen -v`).

**Comment by bollenn** [comment]
I've updated the issue with real version number as per `doxygen -v` and added html output.

Btw it looks like the `stripPath()` can not do its work as `index.cpp` inserts a `index_` which might need to be `index/`

**Comment by albert-github** [comment]
I think there is a small problem with the usage of the `#` here. In HTML this is used for reference into the current page, but it looks like this is not handled properly for other output formats. 
Referencing by means of `#...` is probably in this case not the best choice (it is means here the local "HTML" reference and this is different from the doxygen usage of `#`!). 

It also looks like that the label `header-1` is not handled properly.
I have to dive into the problem a bit.

**Comment by albert-github** [comment]
The reason is, as far as I can quickly see, that the path is necessary when he links go to different files.

**Comment by albert-github** [comment]
Maybe we should interpret the links that start (!) with a `#` as to be seen as `\ref` links.
@doxygen what do you think?

**Comment by albert-github** [comment]
I wanted to look into the "header-1" problem but I now see that there is no "header-1" anchor specified.  The first header has no anchor so here doxygen's internal label is present.

**Comment by bollenn** [comment]
Ok, well I'm used from other markdown interpreters that they automatically add an anchor which is the tittle in lower case and having spaces replaced by `-`. That would become another markdown extension and new ticket?

**Comment by albert-github** [comment]
Remarks on https://github.com/doxygen/doxygen/issues/8956#issuecomment-990654191: That would not work, in doxygen, as in that case we could have the same label when e.g. in 2 different pages there would be the same header text. Doxygen looks at a global level and not just in the local file. I don't think the automatic generation of labels is defined in one of the standards. Probably markdown interpreters think it might be useful for their case, but a user should never use automatically generated IDs as they might be changed by the application without notice (Doxygen and probably the markdown renders won't do this light heartily but sometimes it is necessary).

**Comment by albert-github** [comment]
@bollenn with the "1 issue left..." you mean the handling of the starting `#`, it is still on my radar.

**Comment by bollenn** [comment]
correct

**Comment by albert-github** [comment]
For the remaining problem: I've just pushed a proposed patch, pull request #8961

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by bollenn** [comment]
#8961 fixes the issue but now I'm missing a whitespace after the link when the reference is "inlined"

```md
## Header 2 {#label_header_2}

[Link to header 2](#label_header_2) this is text
```

becomes
```latex
\label{index_label_header_2}%
\Hypertarget{index_label_header_2}%
 \mbox{\hyperlink{index_label_header_2}{Link to header 2}}this is text 
```

**Comment by albert-github** [comment]
I just tried it with the current master (1.9.3 (644607d55307e88ed52ed19469aec6c46ed4819c) and I get:
```
\hypertarget{md_bb_label_header_2b}{}\doxysection{Header 2b}\label{md_bb_label_header_2b}
we refer to section \mbox{\hyperlink{md_bb_label_header_2b}{Link to header 2b}} and this is text
```
so this looks OK to me.

My example: [example2.tar.gz](https://github.com/doxygen/doxygen/files/7719138/example2.tar.gz)

**Comment by bollenn** [comment]
Seems you are correct, somehow this is introduced by https://github.com/doxygen/doxygen/pull/8953, am on it...

So in the end, this issue is fixed on master, lets wait for release

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by bollenn** [comment]
I confirm the issue is resolved in 1.9.3, thx for the support!
